torchaudio/csrc/sox_effects.cpp

```diff
@@ -121,16 +121,17 @@ c10::intrusive_ptr<TensorSignal> apply_effects_file(
   chain.run();
 
   // Create tensor from buffer
+  bool channels_first_ = channels_first.value_or(true);
   auto tensor = convert_to_tensor(
       /*buffer=*/out_buffer.data(),
       /*num_samples=*/out_buffer.size(),
       /*num_channels=*/chain.getOutputNumChannels(),
       dtype,
-      normalize,
-      channels_first);
+      normalize.value_or(true),
+      channels_first_);
```

Comment:
I think you could even do
```
channels_first ? *channels_first : true
```

Reply:
Yeah, it's just the resulting value is used twice, so I had to cache it other wise we define default value twice.